src/components/Api.js

```diff
@@ -0,0 +1,181 @@
+export default class Api {
+  constructor({ baseURL, authToken }) {
+    this._baseURL = baseURL;
+    this._authToken = authToken;
+  }
+
+  // GET https://around.nomoreparties.co/
+  getCardList() {
+    return fetch("https://around.nomoreparties.co/v1/group-12/cards", {
+      headers: {
+        authorization: this._authToken,
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      // if the server returns an error, reject the promise
+      return Promise.reject(`Error: ${res.status}`).catch((err) => {
```

Comment:
Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```